fix: phone-input

hemantmm opened the pull request:
closes: #23722

## What does this PR do?

- Fixes #23722 
- Fixes CAL-6392

This PR consolidates phone input handling by using a single `attendeePhoneNumber` field instead of separate fields for SMS reminder number and AI agent call phone number. Users will now see only one phone input field on booking forms.

### Key Changes:
- Skip rendering legacy phone fields (`smsReminderNumber`, `aiAgentCallPhoneNumber`) in `BookingFields.tsx`
- Consolidate phone number sources in `getBookingFields.ts` to surface a single `attendeePhoneNumber` field
- Preserve user's existing `required` setting when updating the `attendeePhoneNumber` field

## Visual Demo (For contributors especially)

#### Video Demo (if applicable):

Before:

https://github.com/user-attachments/assets/aef4d329-baa0-4642-9ed3-f27c11d9829e

After:

https://github.com/user-attachments/assets/32b0a5b6-37d6-46e9-9cef-3c1c2e5c3676

## Updates since last revision

Completed by Devin to address review feedback and resolve merge conflicts:

1. **Merged with upstream/main** - Resolved conflicts after `.eslintrc.js` was deleted and `BookingFields.tsx` was moved to `apps/web/modules/bookings/components/BookEventForm/`
2. **Fixed `required` property issue** - Changed `required: isPhoneNumberRequired` to `required: bookingFields[attendeePhoneNumberIndex].required || isPhoneNumberRequired` to preserve user's existing configuration (addresses cubic-dev-ai review feedback)
3. **Applied lint fixes** - Import organization cleanup

## Mandatory Tasks (DO NOT REMOVE)

- [x] I have self-reviewed the code (A decent size PR without self-review might be rejected).
- [x] I have updated the developer docs in /docs if this PR makes changes that would require a [documentation change](https://cal.com/docs). If N/A, write N/A here and check the checkbox.
- [x] I confirm automated tests are in place that prove my fix is effective or that my feature works.

## How should this be tested?

1. Create an event type with a workflow that requires SMS notifications or AI phone calls
2. Navigate to the booking page for that event type
3. Verify only one phone number field (`attendeePhoneNumber`) is displayed instead of multiple phone fields
4. If the user had previously set the phone field as required, verify it remains required

## Human Review Checklist

- [ ] Verify the skip logic for legacy phone fields in `apps/web/modules/bookings/components/BookEventForm/BookingFields.tsx` is correct
- [ ] Verify the `required` property preservation logic in `packages/features/bookings/lib/getBookingFields.ts` works as expected
- [ ] Test with workflows that use SMS/WhatsApp attendee notifications and AI phone calls

## Checklist

- I haven't read the [contributing guide](https://github.com/calcom/cal.com/blob/main/CONTRIBUTING.md)
- My code doesn't follow the style guidelines of this project
- I haven't commented my code, particularly in hard-to-understand areas
- I haven't checked if my changes generate no new warnings

---

**Link to Devin run:** https://app.devin.ai/sessions/598091a8c87e4189b79221e96219f8f9
**Requested by:** unknown ()
**Original author:** @hemantmm
<!-- devin-review-badge-begin -->

---

<a href="https://app.devin.ai/review/calcom/cal.com/pull/23729" target="_blank">
  <picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://static.devin.ai/assets/gh-open-in-devin-review-dark.svg?v=1">
    <img src="https://static.devin.ai/assets/gh-open-in-devin-review-light.svg?v=1" alt="Open with Devin">
  </picture>
</a>
<!-- devin-review-badge-end -->

hemantmm committed:
fix: phone-input

closes: #23722

hemantmm committed:
Merge branch 'main' into phone-input-integration

Devanshusharma2005 committed:
Merge branch 'main' into phone-input-integration

hemantmm committed:
Merge branch 'main' into phone-input-integration

hemantmm committed:
Merge branch 'main' into phone-input-integration

hemantmm committed:
Merge branch 'main' into phone-input-integration

hemantmm committed:
Merge branch 'main' into phone-input-integration

kart1ka commented on packages/features/bookings/Booker/components/BookEventForm/BookingFields.tsx:
```diff
@@ -25,21 +23,16 @@ export const BookingFields = ({
   const isInstantMeeting = useBookerStore((state) => state.isInstantMeeting);
 
   return (
-    // TODO: It might make sense to extract this logic into BookingFields config, that would allow to quickly configure system fields and their editability in fresh booking and reschedule booking view
-    // The logic here intends to make modifications to booking fields based on the way we want to specifically show Booking Form
```
why were these comments removed? please restore them if not done for specific reasons.

kart1ka commented on packages/features/bookings/Booker/components/BookEventForm/BookingFields.tsx:
```diff
@@ -25,21 +23,16 @@ export const BookingFields = ({
         if (isInstantMeeting && field.name === "location") return null;
 
         // During reschedule by default all system fields are readOnly. Make them editable on case by case basis.
-        // Allowing a system field to be edited might require sending emails to attendees, so we need to be careful
```
same here

kart1ka commented on packages/features/bookings/Booker/components/BookEventForm/BookingFields.tsx:
```diff
@@ -60,29 +52,22 @@ export const BookingFields = ({
         }
 
-        if (field.name === SystemField.Enum.smsReminderNumber) {
-          // `smsReminderNumber` and location.optionValue when location.value===phone are the same data point. We should solve it in a better way in the Form Builder itself.
-          // I think we should have a way to connect 2 fields together and have them share the same value in Form Builder
-          if (locationResponse?.value === "phone") {
-            setValue(`responses.${SystemField.Enum.smsReminderNumber}`, locationResponse?.optionValue);
-            // Just don't render the field now, as the value is already connected to attendee phone location
-            return null;
-          }
-          // `smsReminderNumber` can be edited during reschedule even though it's a system field
```
same here as well and the rest of the places where they were removed without any reason

kart1ka reviewed (changes requested):
Left some comments. We are going to need tests to verify these changes.

hemantmm committed:
Merge branch 'main' into phone-input-integration

hemantmm committed:
Merge branch 'main' into phone-input-integration

hemantmm committed:
fix: comment issue

hemantmm committed:
Merge branch 'main' into phone-input-integration

hemantmm committed:
Merge branch 'main' into phone-input-integration

hemantmm committed:
Merge branch 'main' into phone-input-integration

hemantmm committed:
fix: typechecks

hemantmm committed:
fix: coderabbit suggestion

Signed-off-by: Hemant M Mehta <hemant29mehta@gmail.com>

hemantmm committed:
fix: unnecessary code

Signed-off-by: Hemant M Mehta <hemant29mehta@gmail.com>

hemantmm committed:
Merge branch 'main' into phone-input-integration

hemantmm committed:
Merge branch 'main' into phone-input-integration

Devanshusharma2005 committed:
Merge branch 'main' into phone-input-integration

Devanshusharma2005 commented on packages/features/bookings/Booker/components/BookEventForm/BookingFields.tsx:
```diff
@@ -33,8 +31,6 @@ export const BookingFields = ({
 }) => {
   const { t, i18n } = useLocale();
-  const { watch, setValue } = useFormContext();
-  const locationResponse = watch("responses.location");
```
Could you please explain me the reason for removing these ?

hemantmm replied:
Unused in the component, neither the watch, setValue, nor locationResponse was referenced anywhere else in the file, keeping them would cause unnecessary variable declarations.

Devanshusharma2005 commented on packages/features/bookings/Booker/components/BookEventForm/BookingFields.tsx:
```diff
@@ -112,19 +107,17 @@ export const BookingFields = ({
           // rescheduleReason is a reschedule specific field and thus should be editable during reschedule
-          readOnly = false;
+          readOnly = false; // rescheduleReason is editable during reschedule
         }
```
lets remove the comments.

Devanshusharma2005 reviewed (commented):
Left some comments.

hemantmm committed:
fix: remove comments

hemantmm committed:
Merge branch 'main' into phone-input-integration

hemantmm committed:
Merge branch 'main' into phone-input-integration

pallava-joshi reviewed (commented):
hey can you please fix the merge conflicts here. marking it draft until.

devin-ai-integration[bot] committed:
Merge upstream/main into phone-input-integration

Co-Authored-By: unknown <>

devin-ai-integration[bot] committed:
fix: consolidate phone input fields and preserve user's required setting

- Skip legacy phone fields (smsReminderNumber, aiAgentCallPhoneNumber) in BookingFields.tsx
- Preserve user's required setting when updating attendeePhoneNumber field
- Apply lint fixes for import organization

Co-authored-by: hemantmm <hemantmm@users.noreply.github.com>
Co-Authored-By: unknown <>

hemantmm committed:
Merge branch 'main' into phone-input-integration

sahitya-chandra committed:
Merge branch 'main' into phone-input-integration

Ryukemeister reviewed (commented):
hi there @hemantmm, thank you for your contribution. this is an important issue, hence we think it's best if the core members work on this. i'll close this PR for now, feel free to pick any of the other issues that haven't been assigned to anyone, cheers!
